Mathlib/FieldTheory/SimpleExtension.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.FieldTheory.PrimitiveElement
+import Mathlib.RingTheory.Henselian
+import Mathlib.Algebra.Lie.OfAssociative
+
+/-!
+# Simple field extensions
+
+We define a typeclass `SimpleExtension` for one field being a simple extension of another
+(this is data because it contains which element was adjoined).
+
+We also define `TranscendentalExtension F K`, and show that it implies that `K` is `F[X]`'s fraction
+field.
+-/
+
+open Polynomial IntermediateField algebraMap
+
+/--
+A typeclass for K being a simple extension of F, F⟮k⟯ = K.
+-/
+class SimpleExtension (F K : Type*) [CommSemiring F] [Field K] [Algebra F K] where
+  /-- The element of K which is adjoined to F. -/
+  k : K
+  /-- Every value in K is a fraction of two polynomials with coefficients in F, evaluated at k. -/
+  is_adjoin' : ∀ (x : K), ∃ (r s : Polynomial F), x = (aeval k) r / (aeval k) s
+
+/--
+We can use `Field.exists_primitive_element` to noncomputable create a simple extension.
+-/
+noncomputable def SimpleExtesion_of_finite_seperable (F K : Type*) [Field F] [Field K] [Algebra F K]
+    [FiniteDimensional F K] [Algebra.IsSeparable F K] : SimpleExtension F K where
+  k := (Field.exists_primitive_element F K).choose
+  is_adjoin' x := by
+    rw [← IntermediateField.mem_adjoin_simple_iff, (Field.exists_primitive_element F K).choose_spec]
+    apply IntermediateField.mem_top
+
+section CommSemiring
+
+variable (F K : Type*) [CommSemiring F] [Field K] [Algebra F K] [se : SimpleExtension F K]
+
+instance (priority := 100) toAlgebra : Algebra F[X] K := (aeval se.k).toAlgebra
+
+lemma SimpleExtension.is_adjoin : ∀ (x : K), ∃ (r s : Polynomial F), x = r / s := se.is_adjoin'
+
+lemma algebraMap_eq_aeval : algebraMap F[X] K = aeval (R := F) se.k := rfl
+
+lemma SimpleExtension.X_eq_k : algebraMap F[X] K X = se.k := aeval_X _
+
+end CommSemiring
+
+section CommRing
+
+variable (F K : Type*) [CommRing F] [Nontrivial F] [IsDomain F] [Field K] [Algebra F K]
+    [se : SimpleExtension F K]
+
+/--
+A typeclass for K being a transcendental simple extension of F.
+-/
+class TranscendentalExtension : Prop where
+  is_transcendental : Transcendental F se.k
```

Comment:
I think your definition means purely transcendental of transcendental degree one, right? But from your name `TranscendentalExtension` neither purely transcendental nor of transcendental degree one can be inferred, which may be misleading.

Reply:
Maybe it's better to just create `Algebra.IsTranscendental` as not `Algebra.IsAlgebraic` which implies this?

Reply:
I mean maybe the name could be changed to `PurelyTranscendentalOfDegOne`

Reply:
I removed this typeclass and replaced this assumptions with `Algebra.Transcendental`, introduced in #14710 .